Sally7/Internal/CancellationTokenExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Sally7.Internal
+{
+    internal static class CancellationTokenExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static CancellationTokenRegistration MaybeUnsafeRegister(this CancellationToken cancellationToken,
+            Action callback)
```

Comment:
The signature should be
```c#
public static CancellationTokenRegistration MaybeUnsafeRegister(this CancellationToken cancellationToken, Action<object?> callback, object? state)
```
in order to avoid several allocations -- here and where it's used.
* the closure (allocation of the "display class")
* the allocation for the repeated creation of the delegate

Reply:
You amaze me again with these details. When you're saying them they make absolute sense, but even when I see the line you comment on I'm still trying to figure what you'll say (before reading your comment).

Reply:
Just look at the generated code, e.g. with ILSpy or any other de-compiler (like sharplab). Then you see the closures, etc.
With the suggestion they're gone and we are back to low allocations and high perf.